fix sentry_helper to not use Hub which is deprecated

CI is failing with a lot of this:

```
self = <sentry_sdk.hub.Hub object at 0x7f0b960c2100>, client_or_hub = None
  scope = None
  
      def __init__(
          self,
          client_or_hub=None,  # type: Optional[Union[Hub, Client]]
          scope=None,  # type: Optional[Any]
      ):
          # type: (...) -> None
  >       warnings.warn(SentryHubDeprecationWarning(), stacklevel=2)
  E       sentry_sdk.hub.SentryHubDeprecationWarning: `sentry_sdk.Hub` is deprecated and will be removed in a future major release. Please consult our 1.x to 2.x migration guide for details on how to migrate `Hub` usage to the new API: https://docs.sentry.io/platforms/python/migration/1.x-to-2.x
```

We need to fix it to stop using Hub.

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -9,7 +9,7 @@ license = {text = "MPLv2"}
 requires-python = ">=3.8"
 dependencies = [
     "attrs>=21.2.0",
-    "sentry-sdk>2",
+    "sentry-sdk>=2",
 ]
 classifiers = [
     "Development Status :: 5 - Production/Stable",
@@ -85,6 +85,7 @@ legacy_tox_ini = """
 [tox]
 isolated_build = True
 envlist =
+    py38-sentry2,
     py{38,39}-django{none,42},
     py310-django{none,42,50},
     py{311,312}-django{none,42,50},
@@ -102,6 +103,7 @@ python =
 
 [testenv]
 deps =
+    sentry2: sentry-sdk==2.0.0
     django42: Django>=4.2,<5.0
     django50: Django>=5.0,<5.1
 extras = dev

--- a/src/fillmore/test.py
+++ b/src/fillmore/test.py
@@ -130,13 +130,17 @@ class SentryTestHelper:
            https://docs.sentry.io/platforms/python/configuration/options/
 
         """
-        with sentry_sdk.Hub(None):
-            hub = sentry_sdk.Hub.current
-            client = sentry_sdk.Client(*args, **kwargs)
-            hub.bind_client(client)
+        scope = sentry_sdk.Scope.get_global_scope()
+        client = sentry_sdk.Client(*args, **kwargs)
+        scope.set_client(client)
 
-            self._transport.reset()
-            client.transport = self._transport
+        self._transport.reset()
+
+        # Clear the breadcrumbs in the scope
+        scope.clear_breadcrumbs()
+
+        # Mock the transport with one that captures events
+        with patch.object(client, attribute="transport", new=self._transport):
             yield self
 
     @contextmanager
@@ -258,7 +262,7 @@ def diff_structure(
     if a is ANY or b is ANY:
         return []
 
-    if type(a) != type(b):
+    if type(a) is not type(b):
         return [
             {
                 "msg": f"different types a:{type(a)} b:{type(b)}",

